components/core/src/clp_s/ReaderUtils.hpp

```diff
@@ -17,7 +17,7 @@ class ReaderUtils {
                 : TraceableException(error_code, filename, line_number) {}
     };
 
-    typedef std::map<int32_t, std::set<int32_t>> SchemaMap;
+    typedef std::map<int32_t, std::vector<int32_t>> SchemaMap;
```

Comment:
If we add a `Schema` class in this PR, why are we still storing a vector here? I understand that when reading from the archive, we no longer need to care about the ordering, but this can be confusing since `Schema` is only used for compression and `std::vector<int32_t>` is for search/decompression. I have tried a little bit and here's my purposal to replace the vector by `Schema` where it's necessary:
1. add `[]` operator to the `Schema` class so that it can be indexed as a vector
2. when reading from the archive to a shema, instead of calling `push_back` we can call `insert_unordered` (since the order doesn't matter)
3. add `data()` method to get the underlying raw vector when passing the schema as a column

Reply:
I made changes to use Schema on the reader side as well, but I didn't end up having to add the `[]` or `data()` methods to get things working. The iterators on Schema satisfied the current uses without any additional code change.

Do you think we should still add these methods in case we want them later, or is this good for now?